Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -1225,6 +1225,12 @@ static String mutationObserverNotificationString()
     return std::nullopt;
 }
 
+std::optional<PDFDocumentLayout::PageIndex> UnifiedPDFPlugin::indexForCurrentPageInView() const
+{
+    auto centerInDocumentSpace = convertFromPluginToDocument(flooredIntPoint(size() / 2));
```

Comment:
I didn't expect us to find the current page index like this

Reply:
I was thinking you'd go off of the context menu event position

Reply:
Wouldn't that be more reliable?

Reply:
That is what I was initially planning on doing but our current behavior seems to go based off the page that is in the center of the view regardless of where the event occurred

Reply:
Ah, I see what you mean.